### PR TITLE
Add linear and quadratic weight options to KappaScore

### PR DESCRIPTION
**Pull request by nswitanek**
The KappaScore metric commonly has linear and quadratically weighted variants.
Like the prior, "unweighted" KappaScore, this PR follows the example of scikit-learn, translating from Numpy to Pytorch operators.

### CODE REVIEW

**Comment by sgugger** [comment]
Please sign the [Contributor Licence Agreement](https://www.clahub.com/agreements/fastai/fastai) before we can merge your PR, thanks!

File: fastai/metrics.py [hunk 1]
```diff
@@ -238,17 +238,34 @@ class KappaScore(ConfusionMatrix):
     """
     Compute the rate of agreement (Cohens Kappa).
     Ref.: https://github.com/scikit-learn/scikit-learn/blob/bac89c2/sklearn/metrics/classification.py
+
```
Comment (sgugger) [inline fastai/metrics.py]: Please put this in the documentation notebook `metrics.ipynb` instead, we like the doc string to be as short as possible (fastai style guide).

File: fastai/metrics.py [hunk 2]
```diff
@@ -238,17 +238,34 @@ class KappaScore(ConfusionMatrix):
...
         sum0 = self.cm.sum(dim=0)
         sum1 = self.cm.sum(dim=1)
         expected = torch.einsum('i,j->ij', (sum0, sum1)) / sum0.sum()
+        if self.weights is None:
+            w = torch.ones((self.n_classes, self.n_classes))
+            w[self.x, self.x] = 0
+        elif self.weights == "linear" or self.weights == "quadratic":
+            w = torch.zeros((self.n_classes, self.n_classes))
+            w += torch.arange(self.n_classes, dtype=torch.float)
+            if self.weights == "linear":
```
Comment (sgugger) [inline fastai/metrics.py]: We can make this in one line: `w = ... if self.weights == "linear" else ...`

File: fastai/metrics.py [hunk 3]
```diff
@@ -238,17 +238,34 @@ class KappaScore(ConfusionMatrix):
...
+                w = torch.abs(w - torch.t(w))
+            else:
+                w = (w - torch.t(w)) ** 2
+        else:
```
Comment (sgugger) [inline fastai/metrics.py]: All on the same line (fastai style guide) and maybe add "expected linear or quadratic, got ..." in the error message.

**Comment by sgugger** [comment]
Thanks! There is a little bit we can refactor, and we should move the docstring in the docs.

File: fastai/metrics.py (hunk 1 above)
Comment (nswitanek) [reply fastai/metrics.py]: I left the first line and moved the sklearn reference to `metrics.ipynb`. For consistency, I did the same with the sklearn reference in `MatthewsCorreff` in `metrics.py`.

I added discussion of parameters and deployment to `metrics.ipynb`.

**Comment by sgugger** [comment]
Thanks a lot!
